### ISSUE TITLE
Country filter is not indented nicely on an iPhone

### ISSUE THREAD

**Issue by cjyabraham** [LF-Engineering/lfevents#300]
reported by @dankohn 
Can this be fixed:

![Image from iOS](https://user-images.githubusercontent.com/463160/66316044-1bcbd500-e941-11e9-90bc-5a5e54a4ffbd.jpg)

**Comment by andycochran** [comment]
I'm not seein' it. What's the issue?

**Comment by dankohn** [comment]
On the web, the country is indented below the continent. On the iPhone, they're all centered, so you lose the hierarchy. 

See https://stackoverflow.com/questions/23883553/center-align-text-in-select-drop-down-for-all-browsers

**Comment by andycochran** [comment]
IDK if we want to get into styling OS-specific UI. The problem comes from the indent being achieved by prefixing countries with `&nbsp;&nbsp;` so that they're indented. But since iOS centers `<select>` options, the spaces don't do anything. 

We could instead achieve even more hierarchy by prefixing the countries with `-&nbsp;&nbsp;` and making continents all caps so that there's even hierarchy when the text is centered. 

![image](https://user-images.githubusercontent.com/409279/67163012-279f9a00-f338-11e9-988a-c82b8d01a64e.png)

![image](https://user-images.githubusercontent.com/409279/67163054-8c5af480-f338-11e9-9f99-940b9989a57c.png)

**Comment by andycochran** [comment]
(Also, that stack overflow article is about the opposite problem: changing all browsers to center select options, which I don't think we should do.)

**Comment by dankohn** [comment]
Your proposed solution LGTM

**Comment by cjyabraham** [comment]
As discussed, I'll submit this upstream to the plugin to try and get it fixed there.

**Comment by cjyabraham** [comment]
I've [submitted this issue upstream](https://wordpress.org/support/topic/iphone-hierarchical-filters-are-problematic/).

**Comment by cjyabraham** [comment]
@deigolocco can you guys think of any solutions to this issue?  (It's ok if you can't)

**Comment by rodmeurer** [comment]
@cjyabraham 

We did some tests here and our suggestion would be to change the library that the plugin uses in the dropdown. It's currently using **chosen**, and it actually doesn't have support for mobile. Therefore, when accessed via cell phone it display the device's native dropdown. Now **select2**, has support for mobile, We set up this environment https://pr-473-lfeventsci.pantheonsite.io to test and it worked well, the style is standardized, both on mobile and on the desktop. This configuration is done by the plugin itself, as shown in the attached image.

![WhatsApp Image 2020-02-11 at 12 22 12](https://user-images.githubusercontent.com/54405402/74246341-20bfdd00-4cc3-11ea-8de3-684ba2c6976f.jpeg)

![lfevent-search-filter-options](https://user-images.githubusercontent.com/54405402/74246343-21f10a00-4cc3-11ea-9738-53d1025b0746.png)

**Comment by dankohn** [comment]
That's good, but can you turn off the box at the top of the dropdown that lets you type characters?

Also, what's the difference in filesize, please?

**Comment by rodmeurer** [comment]
@dankohn @cjyabraham We adjusted via CSS to hide the search field on mobile.
And created a new environment: https://pr-475-lfeventsci.pantheonsite.io/
  PR: https://github.com/LF-Engineering/lfevents/pull/475

About the difference in the size of the files see attached images.

![select2](https://user-images.githubusercontent.com/54405402/74269031-9dfd4900-4ce7-11ea-8042-8c6fb191fdc0.png)

![chosen](https://user-images.githubusercontent.com/54405402/74269038-a05fa300-4ce7-11ea-83ae-46cb8fd78938.png)

**Comment by dankohn** [comment]
That didn't work. I'm still seeing a text box at the top of the dropdown on the new branch.

![image](https://user-images.githubusercontent.com/3083270/74274567-48716e00-4ce0-11ea-99f4-542ee095124d.png)

**Comment by rodmeurer** [comment]
@dankohn it was removed only for mobile version. We'll remove for desktop also.

**Comment by cjyabraham** [comment]
I don't have an iPhone but when I try on Browserstack, I'm seeing both the center-aligned list of countries pop up and the hierarchical list of countries dropdown.  So there are now two ways of choosing the country.  @dankohn do you prefer this to just the single center-aligned list?  And are you sure you'd want to turn off the free-text search box?  I figure when the list of countries gets longer, it'll be more useful.
<img width="318" alt="Screen Shot 2020-02-12 at 10 43 58 AM" src="https://user-images.githubusercontent.com/463160/74301142-ac4d7280-4d84-11ea-95b4-503c21f3aa20.png">

**Comment by dankohn** [comment]
I think we should turn off the free text box and the iPhone select as both are confusing for naive users.

**Comment by cjyabraham** [comment]
@rodmeurer is it possible to hide the iPhone native center-aligned country list?

@dankohn just to confirm, you're wanting to hide the free-text box on both mobile and desktop?  I actually find it super-useful on desktop.

**Comment by dankohn** [comment]
OK, feel free to leave if you really like.
--
Dan Kohn <dan@linuxfoundation.org> +1-415-233-1000
Executive Director, Cloud Native Computing Foundation cncf.io
<https://www.cncf.io/>
dankohn.com <https://www.dankohn.com/> or book on my calendar: dankohn.com/c
<https://www.dankohn.com/c>

**Comment by rodmeurer** [comment]
@cjyabraham we removed the iPhone native center-aligned country list, and left the text box on desktop.

**Comment by cjyabraham** [comment]
Thanks.  Looks good.  Can you fix the [code sniffs](https://app.circleci.com/jobs/github/LF-Engineering/lfevents/6154) please?  LMK if you can't see them.

**Comment by cjyabraham** [comment]
We've decided to hold off on merging this for now.  We did some digging in the analytics and are finding that, of the people that use the country filter, less than 5% are iOS users, so we're not wanting to add more js to the page load that will only be used by those users but slightly slow things down for everyone else.  We also like the current select dropdown better on desktop.  Thanks for coming up with this solution though @rodmeurer.
